zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/InterpreterResultMessage.java

```diff
@@ -16,18 +16,29 @@
  */
 package org.apache.zeppelin.interpreter;
 
+import org.apache.zeppelin.tabledata.ColumnDef;
+
 import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * Interpreter result message
  */
 public class InterpreterResultMessage implements Serializable {
   InterpreterResult.Type type;
   String data;
+  List<ColumnDef.TYPE> columnTypes;
```

Comment:
Is it compatible with the previous InterpreterResult which is save into note.json ? I guess so, but it is better to confirm it.